receiver/filelogreceiver/README.md

```diff
@@ -50,7 +50,8 @@ Tails and parses logs from files.
 | `retry_on_failure.initial_interval` | `1s`                                 | [Time](#time-parameters) to wait after the first failure before retrying.                                                                                                                                                                                       |
 | `retry_on_failure.max_interval`     | `30s`                                | Upper bound on retry backoff [interval](#time-parameters). Once this value is reached the delay between consecutive retries will remain constant at the specified value.                                                                                        |
 | `retry_on_failure.max_elapsed_time` | `5m`                                 | Maximum amount of [time](#time-parameters) (including retries) spent trying to send a logs batch to a downstream consumer. Once this value is reached, the data is discarded. Retrying never stops if set to `0`.     
-| `ordering_criteria.regex`     |                                      | Regular expression used for sorting, should contain a named capture groups that are to be used in `regex_key`.                                                                                                                               |
+| `ordering_criteria.method`     | `regex` |  Specifies the mode to use for ordering. Can be `regex` (sort using regex and sort_by rules) or `mtime` (sort by file modified time). |
```

Comment:
Doesn't this serve basically the same purpose as `ordering_criteria.sort_by.sort_type`? Why not add `mtime` as an additional option there?

This would also allow users to compose multiple types of ordering and filtering.

Reply:
I think I got a little lost looking at the current implementation where every sort type requires the regex to be set, I'll take a stab at reworking to have to be a separate sort type.

Reply:
New implementation for sort_type here: #28850